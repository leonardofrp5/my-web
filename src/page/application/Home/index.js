import React from 'react';

import styles from './style.module.scss';

function Home() {
  return (
    <>
      <div className={styles.container}>
        <h1 className={styles.title}>Leonardo Romero Perez</h1>
      </div>
    </>
  );
}

export default Home;
