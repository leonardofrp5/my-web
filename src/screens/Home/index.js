import { useState } from 'react'

import Footer from 'components/Footer'

import imgProfile from 'assets/Images/imgProfile.png'
import { ReactComponent as IconGitHub } from 'assets/Images/github.svg'
import { ReactComponent as IconTwitter } from 'assets/Images/twitter.svg'
import { ReactComponent as IconInstagram } from 'assets/Images/instagram.svg'
import { ReactComponent as IconLinkedin } from 'assets/Images/linkedin.svg'

import styles from './styles.module.scss'

function Home () {
  const [theme, setTheme] = useState('dark')
  const [statusCheck, setStatusCheck] = useState(false)

  const toggleTheme = () => {
    setStatusCheck(!statusCheck)
    const newTheme = statusCheck ? 'dark' : 'light'
    setTheme(newTheme)
  }

  return (
    <>
      <div className={`container container--${theme}`}>
        <div className={styles.switch}>
          <input
            type='checkbox'
            className={styles.switch__input}
            id='Switch'
            onChange={() => toggleTheme()}
          />
          <label className={styles.switch__label} htmlFor='Switch'>
            <span className={styles.switch__indicator} />
            <span className={styles.switch__decoration} />
          </label>
        </div>
        <img className={`container--${theme}--imgProfile`} src={imgProfile} alt='img_leonardo' />
        <h3 className={`container--${theme}--title`}>Leonardo Romero Perez</h3>
        <div className={styles.containerIcons}>
          <a href='https://github.com/leonardofrp5' target='_blank' rel='noopener noreferrer'>
            <IconGitHub className={`container--${theme}--icon`} />
          </a>
          <a
            href='https://www.linkedin.com/in/leonardo-fabian-romero-perez-516479168/'
            target='_blank'
            rel='noopener noreferrer'
          >
            <IconLinkedin className={`container--${theme}--icon`} />
          </a>
          <a href='https://twitter.com/leofrp' target='_blank' rel='noopener noreferrer'>
            <IconTwitter className={`container--${theme}--icon`} />
          </a>
          <a href='https://www.instagram.com/dev.leofrp/' target='_blank' rel='noopener noreferrer'>
            <IconInstagram className={`container--${theme}--icon`} />
          </a>
        </div>
        <Footer />
      </div>
    </>
  )
}

export default Home
