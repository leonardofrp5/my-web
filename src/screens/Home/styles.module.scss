@import 'scss/variables/colors';

// .container {
//   align-items: center;
//   background-color: $shark;
//   display: flex;
//   flex-direction: column;
//   height: 100vh;
//   justify-content: center;
//   width: 100%;
// }

// .imgProfile {
//   border-radius: 50%;
//   box-shadow: 0 10px 10px -10px $white;
//   height: 25%;
// }

// .title {
//   color: $malibu;
//   font-size: 1.8em;
//   padding: 20px;
//   text-align: center;
// }

// .icon {
//   height: 30px;
//   width: 30px;
//   padding: 0 10px;

//   path {
//     fill: $white;
//   }
// }

.aboutMe {
  width: 40%;
}

.containerIcons {
  padding-top: 20px;
}

.switch {
  display: inline-block;
  margin: 10px;
  position: absolute;
  right: 0;
  top: 0;
}

.switch__input {
  clip: rect(1px, 1px, 1px, 1px);
  clip-path: inset(50%);
  height: 1px;
  width: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
}

.switch__label {
  position: relative;
  display: inline-block;
  width: 40px;
  height: 20px;
  background-color: #2b2b2b;
  border: 5px solid #5b5b5b;
  border-radius: 9999px;
  cursor: pointer;
  transition: all 0.4s cubic-bezier(0.46, 0.03, 0.52, 0.96);
}

.switch__indicator {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%) translateX(-72%);
  display: block;
  width: 13px;
  height: 13px;
  background-color: #7b7b7b;
  border-radius: 9999px;
  box-shadow: 10px 0px 0 0 rgba(#000000, 0.2) inset;

  &::before,
  &::after {
    position: absolute;
    content: '';
    display: block;
    background-color: #ffffff;
    border-radius: 9999px;
  }

  &::before {
    top: 7px;
    left: 7px;
    width: 2px;
    height: 2px;
    background-color: #ffffff;
    opacity: 0.6;
  }

  &::after {
    bottom: 8px;
    right: 6px;
    width: 4px;
    height: 4px;
    background-color: #ffffff;
    opacity: 0.8;
  }
}

.switch__decoration {
  position: absolute;
  top: 65%;
  left: 50%;
  display: block;
  width: 2px;
  height: 2px;
  background-color: #ffffff;
  border-radius: 9999px;
  animation: twinkle 0.8s infinite -0.6s;

  &::before,
  &::after {
    position: absolute;
    display: block;
    content: '';
    width: 2px;
    height: 2px;
    background-color: #ffffff;
    border-radius: 9999px;
  }

  &::before {
    top: -10px;
    left: 10px;
    opacity: 1;
    animation: twinkle 0.6s infinite;
  }

  &::after {
    top: -2px;
    left: 10px;
    animation: twinkle 0.6s infinite -0.2s;
  }
}

@keyframes twinkle {
  50% {
    opacity: 0.2;
  }
}

.switch__indicator {
  &,
  &::before,
  &::after {
    transition: all 0.4s cubic-bezier(0.46, 0.03, 0.52, 0.96);
  }
}

.switch__input:checked + .switch__label {
  background-color: #8fb5f5;
  border-color: #347cf8;

  .switch__indicator {
    background-color: #ecd21f;
    box-shadow: none;
    transform: translate(-50%, -50%) translateX(72%);

    &::before,
    &::after {
      display: none;
    }
  }

  .switch__decoration {
    top: 50%;
    transform: translate(0%, -50%);
    animation: cloud 8s linear infinite;

    width: 6px;
    height: 6px;

    &::before {
      width: 3px;
      height: 3px;
      top: auto;
      bottom: 0;
      left: -3px;
      animation: none;
    }

    &::after {
      width: 5px;
      height: 5px;
      top: auto;
      bottom: 0;
      left: 5px;
      animation: none;
    }

    &,
    &::before,
    &::after {
      border-radius: 9999px 9999px 0 0;
    }

    &::after {
      border-bottom-right-radius: 9999px;
    }
  }
}

@keyframes cloud {
  0% {
    transform: translate(0%, -50%);
  }
  50% {
    transform: translate(-50%, -50%);
  }
  100% {
    transform: translate(0%, -50%);
  }
}
