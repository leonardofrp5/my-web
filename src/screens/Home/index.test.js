import { render } from '@testing-library/react'

import Home from './index'

describe('#Home', () => {
  test('Check component Home', () => {
    const { container } = render(<Home />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
