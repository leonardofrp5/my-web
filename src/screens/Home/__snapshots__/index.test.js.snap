// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Home Check component Home 1`] = `
<div
  class="container container--dark"
>
  <div
    class="switch"
  >
    <input
      class="switch__input"
      id="Switch"
      type="checkbox"
    />
    <label
      class="switch__label"
      for="Switch"
    >
      <span
        class="switch__indicator"
      />
      <span
        class="switch__decoration"
      />
    </label>
  </div>
  <img
    alt="img_leonardo"
    class="container--dark--imgProfile"
    src="imgProfile.png"
  />
  <h3
    class="container--dark--title"
  >
    Leonardo Romero Perez
  </h3>
  <div
    class="containerIcons"
  >
    <a
      href="https://github.com/leonardofrp5"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        class="container--dark--icon"
      >
        github.svg
      </svg>
    </a>
    <a
      href="https://www.linkedin.com/in/leonardo-fabian-romero-perez-516479168/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        class="container--dark--icon"
      >
        linkedin.svg
      </svg>
    </a>
    <a
      href="https://twitter.com/leofrp"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        class="container--dark--icon"
      >
        twitter.svg
      </svg>
    </a>
    <a
      href="https://www.instagram.com/dev.leofrp/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        class="container--dark--icon"
      >
        instagram.svg
      </svg>
    </a>
  </div>
  <footer
    class="footer"
  >
    Fabricated by Leonardo Romero Copyright Â© 
    2021
  </footer>
</div>
`;
